in_mem_metrics: initial documentation.

pwhelan opened the pull request:
This adds documentation for the new `in_mem_metrics` plugin: https://github.com/fluent/fluent-bit/pull/7615.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

* **Documentation**
  * Added documentation for the memory metrics plugin covering configuration parameters (process filters, interval settings, `/proc` path), usage examples for command-line and configuration files with Prometheus remote write support, and complete metrics reference with supported label dimensions.

<!-- review_stack_entry_start -->

[![Review Change Stack](https://storage.googleapis.com/coderabbit_public_assets/review-stack-in-coderabbit-ui.svg)](https://app.coderabbit.ai/change-stack/fluent/fluent-bit-docs/pull/1170?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

<!-- review_stack_entry_end -->

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

pwhelan committed:
in_mem_metrics: initial documentation.

This adds documentation for the new `in_mem_metrics` plugin: https://github.com/fluent/fluent-bit/pull/7615.

Signed-off-by: Phillip Whelan <phil@calyptia.com>

pwhelan commented:
@patrick-stephens should I add a specific example for using docker to log metrics for kubernetes nodes?

patrick-stephens commented:
> @patrick-stephens should I add a specific example for using docker to log metrics for kubernetes nodes?

I don't really understand so probably :)

Although why would you use it over node exporter, kubelet metrics, etc. is probably the main question I'd have? Be good to have an example of what it provides in addition.

pwhelan commented:
> Although why would you use it over node exporter, kubelet metrics, etc. is probably the main question I'd have? Be good to have an example of what it provides in addition.

The main reason to use it over those options is granularity. This plugin grabs the memory usage of each individual process and various different types, ie: shared, etc... If you want to be able to correctly diagnose which process is actually leaking this would be what you need to be able to discern that correctly.

esmerel commented on pipeline/inputs/mem-metrics.md:
```diff
@@ -0,0 +1,110 @@
+# Memory Metrics (mem_metrics)
```
```suggestion
# Memory metrics 
```

esmerel commented on pipeline/inputs/mem-metrics.md:
```diff
@@ -0,0 +1,110 @@
+# Memory Metrics (mem_metrics)
+
+The **mem_metrics**, or memory metrics plugin, generates metrics for memory statistics for processes using the `smaps_rollup` file for each process in /proc. At the moment this plugin works exclusively on Linux.
```
```suggestion
The _mem_metrics_, or memory metrics plugin, generates metrics for memory statistics for processes using the `smaps_rollup` file for each process in /proc. This plugin works exclusively on Linux.
```

esmerel commented on pipeline/inputs/mem-metrics.md:
```diff
@@ -0,0 +1,110 @@
+
+The `filter_pid` can include or be set to either `0` or `self` to refer to the fluent-bit process itself.
+
+## Getting Started
```
```suggestion
## Get started
```

esmerel commented on pipeline/inputs/mem-metrics.md:
````diff
@@ -0,0 +1,110 @@
+    Match mem
+```
+
+Log all instances of fluent-bit to prometheus:
````
```suggestion
Log all instances of Fluent Bit to Prometheus:
```

esmerel reviewed (commented):
Generally stylistic changes. This looks fairly ok otherwise.

eschabell commented:
@pwhelan we have some review suggestions for your attention.

pwhelan committed:
Update pipeline/inputs/mem-metrics.md

Co-authored-by: Lynette  Miles <6818907+esmerel@users.noreply.github.com>
Signed-off-by: Phillip Whelan <pwhelan@exis.cl>

pwhelan committed:
Update pipeline/inputs/mem-metrics.md

Co-authored-by: Lynette  Miles <6818907+esmerel@users.noreply.github.com>
Signed-off-by: Phillip Whelan <pwhelan@exis.cl>

pwhelan committed:
Update pipeline/inputs/mem-metrics.md

Co-authored-by: Lynette  Miles <6818907+esmerel@users.noreply.github.com>
Signed-off-by: Phillip Whelan <pwhelan@exis.cl>

pwhelan committed:
Update pipeline/inputs/mem-metrics.md

Co-authored-by: Lynette  Miles <6818907+esmerel@users.noreply.github.com>
Signed-off-by: Phillip Whelan <pwhelan@exis.cl>

pwhelan committed:
Update pipeline/inputs/mem-metrics.md

Co-authored-by: Lynette  Miles <6818907+esmerel@users.noreply.github.com>
Signed-off-by: Phillip Whelan <pwhelan@exis.cl>

pwhelan committed:
Update pipeline/inputs/mem-metrics.md

Co-authored-by: Lynette  Miles <6818907+esmerel@users.noreply.github.com>
Signed-off-by: Phillip Whelan <pwhelan@exis.cl>

pwhelan committed:
Update pipeline/inputs/mem-metrics.md

Co-authored-by: Lynette  Miles <6818907+esmerel@users.noreply.github.com>
Signed-off-by: Phillip Whelan <pwhelan@exis.cl>

pwhelan committed:
Update pipeline/inputs/mem-metrics.md

Co-authored-by: Lynette  Miles <6818907+esmerel@users.noreply.github.com>
Signed-off-by: Phillip Whelan <pwhelan@exis.cl>

eschabell committed:
docs: pipeline: inputs: mem-metrics: fix formatting and style issues

  - Fix 'Fluent Bit' casing (was 'fluent-bit') in body text
  - Sort configuration parameters table alphabetically
  - Fix classic config block indentation from 4 spaces to 2 spaces
  - Fixed header for config param table

  Applies to #1170

Signed-off-by: Eric D. Schabell <eric@schabell.org>

eschabell commented on pipeline/inputs/mem-metrics.md:
```diff
@@ -0,0 +1,114 @@
+
+# Configuration parameters
+
+| **Key***      | Description                                        | Default      |
```
Fixing.

eschabell reviewed (approved):
@pwhelan this is now waiting on code PR merging.
